pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,512 @@
+	edgev1alpha1informers "github.com/kubestellar/kubestellar/pkg/generated/informers/externalversions/edge/v1alpha1"
+	edgev1alpha1listers "github.com/kubestellar/kubestellar/pkg/generated/listers/edge/v1alpha1"
+)
+
+const (
+	ControllerName                 = "transport-controller"
+	transportFinalizer             = "transport.kubestellar.io/object-cleanup"
+	notFoundErrorSuffix            = "not found"
+	originOwnerReferenceAnnotation = "transport.kubestellar.io/originOwnerReferencePlacementDecisionKey"
+	originWdsAnnotation            = "transport.kubestellar.io/originWdsName"
+)
+
+// NewTransportController returns a new transport controller
+func NewTransportController(ctx context.Context, placementDecisionInformer edgev1alpha1informers.PlacementDecisionInformer, transport Transport,
+	wdsClientset *edgeclientset.Clientset, wdsDynamicClient *dynamic.DynamicClient, transportRestConfig *rest.Config, wdsName string) (*genericTransportController, error) {
+	transportDynamicClient, err := dynamic.NewForConfig(transportRestConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create dynamic k8s clientset for transport space - %w", err)
+	}
+
+	emptyWrappedObject := transport.WrapObjects(make([]*unstructured.Unstructured, 0)) // empty wrapped object to get GVR from it.
+	wrappedObjectGVR, err := getGvrFromWrappedObject(transportRestConfig, emptyWrappedObject)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get transport wrapped object GVR - %w", err)
+	}
+
+	dynamicInformer := dynamicinformer.NewDynamicSharedInformerFactory(transportDynamicClient, 0)
+	wrappedObjectGenericInformer := dynamicInformer.ForResource(*wrappedObjectGVR)
+
+	transportController := &genericTransportController{
+		logger:                          klog.FromContext(ctx).WithName(ControllerName),
+		placementDecisionLister:         placementDecisionInformer.Lister(),
+		placementDecisionInformerSynced: placementDecisionInformer.Informer().HasSynced,
+		wrappedObjectInformerSynced:     wrappedObjectGenericInformer.Informer().HasSynced,
+		workqueue:                       workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName),
+		transport:                       transport,
+		transportClient:                 transportDynamicClient,
+		wrappedObjectGVR:                *wrappedObjectGVR,
+		wdsClientset:                    wdsClientset,
+		wdsDynamicClient:                wdsDynamicClient,
+		wdsName:                         wdsName,
+	}
+
+	transportController.logger.Info("Setting up event handlers")
+	// Set up an event handler for when PlacementDecision resources change
+	placementDecisionInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    transportController.enqueuePlacementDecision,
+		UpdateFunc: func(_, new interface{}) { transportController.enqueuePlacementDecision(new) },
+		DeleteFunc: transportController.enqueuePlacementDecision,
```

Comment:
The DeleteFunc needs to be able to handle a `cache.DeletedFinalStateUnknown` (see https://github.com/kubernetes/client-go/blob/v0.28.2/tools/cache/controller.go#L213-L216).

Reply:
done.
handling function is now using `cache.DeletionHandlingMetaNamespaceKeyFunc`